code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -578,11 +578,336 @@
+		if(get_dist(src,target) <= 3)//If they're close enough just aim straight at them so we don't miss at point blank ranges
+			Shoot(target)
+			return
+		var/turf/our_turf = get_turf(src)
+		var/obj/item/projectile/seedling/readied_shot = new /obj/item/projectile/seedling(our_turf)
+		readied_shot.current = our_turf
+		readied_shot.starting = our_turf
+		readied_shot.firer = src
+		readied_shot.original = target
+		readied_shot.yo = target.y - our_turf.y + rand(-1,1)
+		readied_shot.xo = target.x - our_turf.x + rand(-1,1)
+		readied_shot.fire()
+		playsound(src, projectilesound, 100, 1)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/AttackRecovery()
+	if(combatant_state == SEEDLING_STATE_ACTIVE)
+		combatant_state = SEEDLING_STATE_RECOVERY
+		update_icons()
+		ranged_cooldown = world.time + ranged_cooldown_time
+		if(target)
+			face_atom(target)
+		if(solar_beam)
+			qdel(solar_beam)
+			solar_beam = null
+		addtimer(CALLBACK(src, .proc/ResetNeutral), 10)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/ResetNeutral()
+	if(combatant_state != SEEDLING_STATE_REGENERATING)
+		combatant_state = SEEDLING_STATE_NEUTRAL
+		if(target && !stat)
+			update_icons()
+			Goto(target, move_to_delay, minimum_distance)
+
+/mob/living/simple_animal/hostile/jungle/seedling/adjustHealth()
+	. = ..()
+	if(combatant_state == SEEDLING_STATE_ACTIVE && beam_debuff_target)
+		beam_debuff_target.remove_status_effect(/datum/status_effect/seedling_beam_indicator)
+		beam_debuff_target = null
+		AttackRecovery()
+	if(health <= 0)
+		if(combatant_state == SEEDLING_STATE_REGENERATING)
+			return
+		if(weak_point && !stat)
+			SeedlingStartRegeneration()
+
+/mob/living/simple_animal/hostile/jungle/seedling/update_stat()
+	if(combatant_state == SEEDLING_STATE_REGENERATING)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/SeedlingStartRegeneration()
+	weak_point.loc = get_turf(src)
+	weak_point.RandomizeScreenLocation()
+	combatant_state = SEEDLING_STATE_REGENERATING
+	mouse_opacity = 0
+	update_icons()
+	walk(src,0)
+	wander = FALSE
+	addtimer(CALLBACK(src, .proc/SeedlingRegenerate), SEEDLING_REGENERATION_TIME)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/SeedlingRegenerate()
+	if(weak_point && !QDELETED(weak_point) && combatant_state == SEEDLING_STATE_REGENERATING)
+		revive(TRUE)
+		mouse_opacity = 1
+		weak_point.loc = src
+		wander = TRUE
+		combatant_state = SEEDLING_STATE_NEUTRAL
+		ResetNeutral()
+
+/obj/seedling_weakpoint
+	name = "seedling weakpoint"
+	anchored = TRUE
+	max_integrity = 45
+	icon = 'icons/mob/jungle/arachnid.dmi'
+	icon_state = "seedling_weakpoint1"
+	pixel_x = -16
+	pixel_y = -14
+	layer = ABOVE_ALL_MOB_LAYER
+	var/mob/living/simple_animal/hostile/jungle/seedling/our_seedling
+
+/obj/seedling_weakpoint/Destroy()
+	. = ..()
+	if(our_seedling && !QDELETED(our_seedling))
+		if(!our_seedling.stat)
+			our_seedling.combatant_state = SEEDLING_STATE_NEUTRAL
+			our_seedling.death()
+
```

Comment:
ergo, it will *TRY* to qdel itself after this finishes, btu since our_seedling is still pointing to the mob, there's a reference, and it won't qdel properly, so it'll sit in the qdel queue for a little while and THEN it'll trigger a harddel (a ```del()``` which is stupidly bad)

Mobs already don't qdel well, but I think simple animals can, so don't make it worse.